### DISCUSSION TITLE
hello, everyone, I create a code generator for this CleanArchitecture project

### DISCUSSION THREAD

**Discussion by neozhu** [jasontaylordev/CleanArchitecture#635] [General]
[![cool!](https://i9.ytimg.com/vi/Hp6cjdfgMT8/mq2.jpg?sqp=CNSu8ogG&rs=AOn4CLDcYRu3HEChyr5fZu3wAL2_n_50Ag)](https://www.youtube.com/watch?v=Hp6cjdfgMT8)

https://github.com/neozhu/CleanArchitectureCodeGenerator

Easily create following application features code
{nameofPlural}/Commands/AddEdit/AddEdit{name}Command.cs
{nameofPlural}/Commands/AddEdit/AddEdit{name}CommandValidator.cs
{nameofPlural}/Commands/Create/Create{name}Command.cs
{nameofPlural}/Commands/Create/Create{name}CommandValidator.cs
{nameofPlural}/Commands/Delete/Delete{name}Command.cs
{nameofPlural}/Commands/Delete/Delete{name}CommandValidator.cs
{nameofPlural}/Commands/Update/Update{name}Command.cs
{nameofPlural}/Commands/Update/Update{name}CommandValidator.cs
{nameofPlural}/Commands/Import/Import{name}Command.cs
{nameofPlural}/Commands/Import/Import{name}CommandValidator.cs
{nameofPlural}/DTOs/{name}Dto.cs
{nameofPlural}/EventHandlers/{name}CreatedEventHandler.cs
{nameofPlural}/EventHandlers/{name}UpdatedEventHandler.cs
{nameofPlural}/EventHandlers/{name}DeletedEventHandler.cs
{nameofPlural}/Queries/Export/Export{nameofPlural}Query.cs
{nameofPlural}/Queries/GetAll/GetAll{nameofPlural}Query.cs
{nameofPlural}/Queries/Pagination/{nameofPlural}PaginationQuery.cs

## It is still in the beginning, and I hope more excellent people will join in the development.

**Comment by jasontaylordev** [comment]
Awesome, I'll give it a try ❤

**Comment by jasontaylordev** [comment]
Thanks for sharing, I gave it a quick test run and it doesn't seem to pick up any models so I couldn't try it out.
I like the initial idea but I think it generates too much - each time I use it I will have to delete many of the files. Suggest sticking with GetAll, GetById, Create, Update and Delete. Or, alternatively, providing some options to let the developer choose what to generate.

**Reply by MattBance** [reply]
I'm not a web dev but.

Could a Code Generator concept like this be used to generate boilerplate code generation like RedWoodJS? [https://github.com/redwoodjs/redwood](url)

Having the two separate would keep everyone happy. Those that want complete control, just use the template. Those that want to trust your opinionated conventions and library choices (me!) can utilise a cli code generator 

What i like about RedwoodJs , is they hide all the library, solution implementation / integration
